server/src/main/scala/services/graphstages/SimulationGraphStage.scala

```diff
@@ -341,6 +341,21 @@ class SimulationGraphStage(name: String = "",
 
     var deploymentCache: Map[Int, Seq[(String, Int)]] = Map()
 
+    @scala.annotation.tailrec
+    def addOneDeskToQueueAtIndex(deployments : List[(String, Int, Int)], index: Int, numberOfQueues: Int, staffAvailable: Int): List[(String, Int, Int)] = {
+      val safeIndex = if (index > numberOfQueues - 1) 0 else index
+      val deployedStaff = deployments.map(_._2).sum
+      val maxStaff = deployments.map(_._3).sum
+      val freeStaff = staffAvailable - deployedStaff
+
+      if (deployedStaff != maxStaff && freeStaff > 0) {
+        val (queue, dr, maxStaff) = deployments(safeIndex)
```

Comment:
dr could probably have a better name, I'm guessing it's desk rec but I'm not sure.